.changeset/config.json

```diff
@@ -0,0 +1,22 @@
+{
+  "$schema": "https://unpkg.com/@changesets/config@2.0.0/schema.json",
+  "changelog": [
```

Comment:
I like how changesets helps with changelogs, and I think we should setup [their bot|https://github.com/apps/changeset-bot] so we have a sure way to not miss changlogs anymore.

That said:

I think we need a little bit of work to bootstrap this correctly. As of now, if one tries a 2.3.0 release from this branch, changelogs ends up looking something like this:
```markdown

## 2.3.0
### Patch Changes



- [#1936](https://github.com/apollographql/federation/pull/1936) [`f2a5b278`](https://github.com/apollographql/federation/commit/f2a5b278ee7c317569ab8d64c741b79ac18b0480) Thanks [@benweatherman](https://github.com/benweatherman)! - Add `changesets` for release management and changelog automation

- Updated dependencies [[`f2a5b278`](https://github.com/apollographql/federation/commit/f2a5b278ee7c317569ab8d64c741b79ac18b0480)]:
  - @apollo/composition@2.3.0
  - @apollo/federation-internals@2.3.0
  - @apollo/query-planner@2.3.0

This CHANGELOG pertains only to Apollo Federation packages in the 2.x range. The Federation v0.x equivalent for this package can be found [here](https://github.com/apollographql/federation/blob/version-0.x/gateway-js/CHANGELOG.md) on the `version-0.x` branch of this repo.

## vNext

## 2.3.0

- Adds support for `@interfaceObject` and keys on interfaces [PR #2279](https://github.com/apollographql/federation/pull/2279).
- Preserves source of union members and enum values in supergraph [PR #2288](https://github.com/apollographql/federation/pull/2288).

## 2.2.2

- Fix issue with path in query plan's deferred nodes [PR #2281](https://github.com/apollographql/federation/pull/2281).
...
```
And ignoring for a minute that my OCD is triggered by changesets adding more newlines than is reasonable, this doesn't look quite right for a bunch of reasons:
1. it obviously doesn't merge our pre-existing 2.3.0 section, so it's duplicated.
2. we have that `vNext` that changesets does not know how to handle and should just go away.
3. changesets doesn't seem to like our "intro" info, the lines that starts with "This CHANGELOG pertains only to Apollo Federation packages in the 2.x range. The Federation v0.x equivalent ...". I scanned the doc and didn't saw an easy way to deal with that, but I personally don't mind tremendously if we just get rid of this (essentially, I think the benefits of having a cleaner handling of changelogs outweighs the benefits of this small line).

Anyway, I think we should either:
1. clean up the changelog now (remove `vNext` and "intro line" everywhere) _and_ retro-fit changesets for the pre-existing changes of 2.3.0.
2. set `changelog: false` here temporarily and release 2.3.0 using our old manual changelog handling. Then just after that, remove `vNext` and the intro line and switch this back from that point one (with the changeset bot setup too).

I'd go with option 2 because it's probably "wiser" to handle the small process change after a release, it gives us a bit of time to figure how to setup the related bot, and it save us from having to retro-fit existing 2.3 stuffs into changesets (not that it's super hard but ...).

Lastly, fwiw, I'm not a super fan of the changelog format of `@changesets/changelog-github`. I do feel listing PR/commit first make reading the changelog harder (especially when reading it in a text editor, which I happen to personally do more often than not), but I'm also not convinced with the `Thanks <author>` for every entry. Don't get me wrong, thanking contributers is great, but as of today, the vast majority of changelog entries are from a handful of Apollo people, and I'm uncomfortable having those same named repeated way too often and polluting the changelog.

Long story short, I'd personally much prefer something like [vitejs/changesets-changelog-github-compact](https://github.com/svitejs/changesets-changelog-github-compact/blob/main/packages/changesets-changelog-github-compact/README.md) (but it seems easy enough to tweak if we really wanted to).

Reply:
set `changelog: false` for now. Totally agree that the format is ugly. Will definitely check out changesets-changelog-github-compact